server/room.ts

```diff
@@ -1,10 +1,13 @@
-import { assignVM, resetVM } from './vm';
 import { Socket } from 'socket.io';
-import { User, ChatMessage, NumberDict, StringDict } from '.';
+
+import { fetchYoutubeVideo, getYoutubeVideoID } from './utils/youtube';
+import { assignVM, resetVM } from './vm';
+import { ChatMessage, NumberDict, PlaylistVideo, StringDict, User } from '.';
 
 export class Room {
-  private video = '';
+  private video?: PlaylistVideo | string;
```

Comment:
Could we maybe keep this as a string and just build a lookup map of strings to playlistvideos to resolve additional metadata?

It was somewhat intentional to represent screenshare and vbrowser state as string to allow a lot of the code around that that to be reused.